test/functional/apps/visualize/_tsvb_markdown.ts

```diff
@@ -71,6 +67,70 @@ export default function({ getPageObjects }: FtrProviderContext) {
         const markdownText = await visualBuilder.getMarkdownText();
         expect(markdownText).to.be(expectedRenderer);
       });
+
+      it('should change label name', async () => {
+        const BASE_LABEL = 'count';
+        const LABEL = 'label';
+        await visualBuilder.markdownSwitchSubTab('data');
+
+        await visualBuilder.setMarkdownDataVariable(LABEL, LABEL);
+
+        await visualBuilder.markdownSwitchSubTab('markdown');
+        const table = await visualBuilder.getMarkdownTableVariables();
+        table.forEach(row => {
+          expect(row.key).to.contain(LABEL);
+        });
+
+        // cleanup label
+        await visualBuilder.markdownSwitchSubTab('data');
+        await visualBuilder.setMarkdownDataVariable('', LABEL);
+
+        await visualBuilder.markdownSwitchSubTab('markdown');
+        const rerenderedTable = await visualBuilder.getMarkdownTableVariables();
+        rerenderedTable.forEach(row => {
+          expect(row.key).to.include.string(BASE_LABEL);
+        });
+      });
+
+      it('should change variable name', async () => {
+        const VARIABLE = 'variable';
+        await visualBuilder.markdownSwitchSubTab('data');
+        await visualBuilder.setMarkdownDataVariable(VARIABLE, VARIABLE);
+        await visualBuilder.markdownSwitchSubTab('markdown');
+        const table = await visualBuilder.getMarkdownTableVariables();
+
+        table.forEach((row, index) => {
+          // exception: last index for variable is always: {{count.label}}
+          index === table.length - 1
+            ? expect(row.key).to.not.include.string(VARIABLE)
+            : expect(row.key).to.include.string(VARIABLE);
+        });
+
+        // cleanup variable name
+        await visualBuilder.markdownSwitchSubTab('data');
+        await visualBuilder.setMarkdownDataVariable('', VARIABLE);
+
+        // table should not be containing variable sault
+        await visualBuilder.markdownSwitchSubTab('markdown');
+        const rerenderedTable = await visualBuilder.getMarkdownTableVariables();
+        rerenderedTable.forEach(row => {
+          expect(row.key).to.not.include.string(VARIABLE);
+        });
```

Comment:
As far as I see, cleanup has common flow for both label & variable, how about putting it in a function and shorter tests a bit:

```
const cleanupMardownData = async (variableName, checkedValue) => {
        await visualBuilder.markdownSwitchSubTab('data');
        await visualBuilder.setMarkdownDataVariable('', variableName);

        await visualBuilder.markdownSwitchSubTab('markdown');
        const rerenderedTable = await visualBuilder.getMarkdownTableVariables();
        rerenderedTable.forEach(row => {
         variableName === 'label' 
           ? expect(row.key).to.include.string(checkedValue)
           : expect(row.key).to.not.include.string(checkedValue);
        });
}
```